### PR TITLE
Merge plotspecutils

### PR DESCRIPTION
**Pull request by armengau**
Not fully ready yet for merge, but ready for review.

`viewer.py` is the result of a manual merge between `plotframes.py` and `plotspecutils.py`.
I kept the most up-to-date code from both, with minimal changes.

Input spectra in viewer.plotspectra() can be Spectra or list[Frame] or Spectrum1D or SpectrumList.
Key changes are at the beginning of plotspectra(), and in make_cds_spectra().

As of now I made tests using *DESI* spectra only:
- from a notebook.
- reprocessing a sample of Blanc VI dataset.
- Still TODO: more tests with DESI spectra (test all options in viewer.plotspectra()).

@weaverba137 Could you in particular:
- look a the code (diff viewer.py plotspecutils.py)
- test on SDSS (Spectrum1D and SpectrumList): I hope it works, but definitely has to be tested.
- Do you have a better idea than `viewer.py` for the name of the file? From the user point of view, `from prospect import viewer` and `viewer.plotspectra()` look ~ok.
- After all tests and before merging, if you agree I can remove `plotframes.py` and `plotspecutils.py`

### CODE REVIEW

**Comment by weaverba137** [comment]
I won't have time to review until next week, but thank you for getting this out.

**Comment by weaverba137** [comment]
With a minor typo fix, the `viewer.py` code is working in a Data Lab notebook.  However there is still an issue with the units of the model spectrum which is preventing plotting SDSS models.  That won't be hard to fix, I've seen it before.

I still want to do a line-by-line code comparison to make sure nothing subtle has been missed, but overall, I think things are working.

**Comment by weaverba137** [comment]
The problem with units on the model is fixed on the Data Lab side, it doesn't need a fix in this code.

**Comment by armengau** [comment]
About flux units: I suppose that flux units will never change in DESI.
I had already noticed that, while the unit is present in the fits file:
eg ``fits.getheader('/global/cfs/cdirs/desi/spectro/redux/daily/tiles/68000/20200314/coadd-0-68000-20200314.fits','brz_flux')['bunit']``
 we apparently cannot recover it from the Spectra objects.
Well, that would be a request to desispec: in desispec.io.read_spectra() put flux units somewhere in meta or extra?

**Comment by weaverba137** [comment]
In this particular case, it's not a case of whether the flux array has (Astropy) units, but what scaling is applied to those units.  Prospect expects `1e-17 erg cm^-2 s^-1 A^-1`, but the Data Lab spectrum service returns `erg cm^-2 s^-1 A^-1`, so it's just a matter of multiplying or dividing by 1e-17 as needed.

File: py/prospect/plotspecutils.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 See :mod:`prospect.specutils` for objects and IO routines.
 """
 import os, sys
-import argparse
```
Comment (weaverba137) [inline py/prospect/plotspecutils.py:12]: I removed this because it was not used.

File: py/prospect/plotframes.py [hunk 2]
```diff
@@ -458,7 +459,7 @@ def plotspectra(spectra, nspec=None, startspec=None, zcatalog=None, redrock_cat=
         num_approx_fits=(nb of best fits available in redrock_cat)
     vi_countdown: if >0, add a countdown widget in the VI panel, with vi_countdown = countdown value in minutes.
     '''
-
+    warnings.warn("prospect.plotframes.plotspectra() is deprecated.  Use prospect.viewer.plotspectra() instead.", DeprecationWarning)
```
Comment (weaverba137) [inline py/prospect/plotframes.py:462]: I added a deprecation warning so we can continue to use, but discourage using, these modules while transitioning.

**Review by weaverba137 — APPROVED** [review]
I think everything is working so far.  I made a few changes already and I think this can be merged as is.

I'm pretty sure `grid_spectra()` will not work with all types of spectra inputs, but that can be a separate PR.

**Pull request merged**